server/plugin/api_test.go

```diff
@@ -295,6 +321,65 @@ func TestHandleLink(t *testing.T) {
 	}
 }
 
+func TestHandleDeleteAllSubscriptions(t *testing.T) {
+	defer monkey.UnpatchAll()
+	mockAPI := &plugintest.API{}
+	mockCtrl := gomock.NewController(t)
+	mockedClient := mocks.NewMockClient(mockCtrl)
+	mockedStore := mocks.NewMockKVStore(mockCtrl)
+	p := setupMockPlugin(mockAPI, mockedStore, mockedClient)
+	for _, testCase := range []struct {
+		description           string
+		userID                string
+		projectID             string
+		err                   error
+		statusCode            int
+		getAllSubscriptionErr error
+		subscriptionList      []*serializers.SubscriptionDetails
+		subscription          *serializers.SubscriptionDetails
+	}{
+		{
+			description: "HandleDeleteAllSubscriptions: valid",
+			userID:      "mockMattermostUserID",
+			projectID:   "mockProjectID",
+			statusCode:  http.StatusOK,
+			subscriptionList: []*serializers.SubscriptionDetails{
+				{
+					MattermostUserID: "mockMattermostUserID",
+					ProjectID:        "mockProjectID",
+					OrganizationName: "mockOrganization",
+					EventType:        "mockEventType",
+					ChannelID:        "mockChannelID",
+					SubscriptionID:   "mockSubscriptionID",
+				},
+			},
+		},
+		{
+			description:           "HandleDeleteAllSubscriptions: GetAllSubscriptions gives error",
+			userID:                "mockMattermostUserID",
+			projectID:             "mockProjectID",
+			statusCode:            http.StatusInternalServerError,
+			getAllSubscriptionErr: errors.New("mockError"),
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			mockAPI.On("LogError", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"))
+
+			if testCase.statusCode == http.StatusOK || testCase.statusCode == http.StatusInternalServerError {
```

Comment:
i think in the above test cases  there are only 2 status codes and both are in this condition so there is no need of this condition